### PR TITLE
charmc: allow extra options to override standard options

### PR DESCRIPTION
**Pull request by matthiasdiener**
Allows specifying (e.g.) -std=c++14 meaningfully

### CODE REVIEW

File: src/scripts/charmc [hunk 1]
```diff
@@ -946,7 +948,7 @@ do
 
 	-*|+*)
 #		Warning "passing unrecognized option $arg to all compilers and linkers"
-		OPTS="$OPTS  $arg"
+        OPTS_EXTRA+=" $arg"
```
Comment (evan-charmworks) [inline src/scripts/charmc:951]:
```suggestion
		OPTS_EXTRA+=" $arg"
```

File: src/scripts/charmc [hunk 2]
```diff
@@ -957,7 +959,7 @@ do
 	*)
 #		printUsage  "Error: Unrecognized argument $arg"
 #		Warning "passing unrecognized option $arg to all compilers and linkers"
-		OPTS="$OPTS  $arg"
+        OPTS_EXTRA+=" $arg"
```
Comment (evan-charmworks) [inline src/scripts/charmc:962]:
```suggestion
		OPTS_EXTRA+=" $arg"
```

**Comment by evan-charmworks** [comment]
This PR changes charmc behavior so that arbitrary options passed to charmc take precedence over arguments specified with `-cc-option` et al, instead of the other way around as it is currently. Would it be possible to append ARG_OPTS_* to OPTS_* after OPTS_EXTRA instead of before it?

**Review by rbuch — APPROVED** [review]

**Comment by matthiasdiener** [comment]
> This PR changes charmc behavior so that arbitrary options passed to charmc take precedence over arguments specified with `-cc-option` et al, instead of the other way around as it is currently. Would it be possible to append ARG_OPTS_* to OPTS_* after OPTS_EXTRA instead of before it?

I genuinely don't know which way is better here. Are there any known users of `-cc-option` etc.?

**Review by evan-charmworks — APPROVED** [review]

**Pull request merged**